src/NuGet.Core/NuGet.Configuration/Settings/Items/SourceItem.cs

```diff
@@ -23,13 +23,41 @@ public string ProtocolVersion
             set => AddOrUpdateAttribute(ConfigurationConstants.ProtocolVersionAttribute, value);
         }
 
-        public SourceItem(string key, string value, string protocolVersion = "")
+        public string AllowInsecureConnections
```

Comment:
You should check what happens when the nuget.config already has a value for `allowInsecureConnections`, and then you edit sources in VS's options page. I think VS will strip out the `allowInsecureConnections` value, so I think you need to add this property into `PackageSourceContextInfo`, its MsgPack formatter, and possibly `NuGetSourcesService.GetPackageSourcesToUpdate`, as I did in this PR: https://github.com/NuGet/NuGet.Client/pull/5334

Reply:
Thanks! That's very helpful! Fixed and added tests. Manually tested in VS, the non-default value of  `allowInsecureConnections` will not be stripped out when updating in VS options page. The default value(false) and invalid value(e.g. "random", will be changed into default value) will be removed after updating, as an expected normalization work. (Just like the protocal Version "2" will be stripped out after updating).